src/OrchardCore.Modules/OrchardCore.Queries/Controllers/ApiController.cs

```diff
@@ -47,8 +49,24 @@ IQueryManager queryManager
                 JsonConvert.DeserializeObject<Dictionary<string, object>>(parameters)
                 : new Dictionary<string, object>();
 
-            var result = await _queryManager.ExecuteQueryAsync(query, queryParameters);
-            return new ObjectResult(result.Items);
+            try
+            {
+                var result = await _queryManager.ExecuteQueryAsync(query, queryParameters);
+                if (withTotal ?? false && result is LuceneQueryResults)
```

Comment:
I believe we should just return the `result` as it in all cases, or have an interface that is common to all results from queries to return the value from this controller.

The fact we see `LuceneQueryResults` means that there is an unnecessary dependency on the Lucene module.